### PR TITLE
Bump Rust toolchain to `nightly-2022-01-20`.

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Bumps the Rust toolchain to `nightly-2022-01-20`.

### Important Notes

New lints about `js-sys` potentially breaking in the future required an upgrade of `wasm-bindgen`. Thus, it was upgraded from `0.2.58` to `0.2.78`. See `app/gui/Cargo.toml` for notes on the upgrade. 

It also includes a bump of `wasm-bindgen` to `0.2.78`.

[ci no changelog needed]

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/develop/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/develop/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: lib/rust/ensogl/example/Cargo.toml [hunk 1]
```diff
@@ -36,7 +36,7 @@ ensogl-example-text-area = { path = "text-area" }
 #ensogl-text-msdf-sys = { path = "../component/text/msdf-sys" }
 #ensogl-hardcoded-theme = { path = "../app/theme/hardcoded" }
 #ensogl-drop-manager = { path = "../component/drop-manager" }
-#wasm-bindgen = { version = "=0.2.58", features = [
+#wasm-bindgen = { version = "0.2.58", features = [
```
Comment (farmaazon) [inline lib/rust/ensogl/example/Cargo.toml]: These commented entries were overlooked and should be removed.

File: app/gui/Cargo.toml [hunk 2]
```diff
@@ -47,7 +47,10 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = { version = "1.0" }
 sha3 = { version = "0.8.2" }
 uuid = { version = "0.8", features = ["serde", "v4", "wasm-bindgen"] }
-wasm-bindgen = { version = "=0.2.58" }
+# wasm-bindgen version 0.2.78 is causing issues with clippy.
+# See for more information. https://github.com/rustwasm/wasm-bindgen/issues/2774
+# Should be removed once 0.2.80 is available.
+wasm-bindgen = { version = "=0.2.78" }
```
Comment (farmaazon) [inline app/gui/Cargo.toml]: I'm a bit confused. We restrict the `wasm-bidgen` version here, but comments says that _this_ version is causing issues. ~~Nevertheless, the `Cargo.lock` has the old version locked only. How does it work?~~

Also, the new version of `wasm-bindgen` may cause the problems in our build scripts, as we patch the output WASM (See `js_workaround_patcher` in run.js file).

**Review by farmaazon — CHANGES_REQUESTED** [review]

File: app/gui/Cargo.toml (hunk 2 above)
Comment (MichaelMauderer) [reply app/gui/Cargo.toml]: Good point! This needs updating.

**Review by wdanilo — APPROVED** [review]

File: app/gui/Cargo.toml (hunk 2 above)
Comment (MichaelMauderer) [reply app/gui/Cargo.toml]: This also fails CI. For some reason, it did not appear locally.

Comment (farmaazon) [reply app/gui/Cargo.toml]: Actually, I have looked at the wrong version of `Cargo.lock`. On this branch, the `wasm-bindgen` is in the new version everywhere, so I'm no longer confused.

File: app/ide-desktop/lib/content/src/index.ts [hunk 3]
```diff
@@ -99,7 +99,7 @@ async function download_content(config: { wasm_glue_url: RequestInfo; wasm_url:
     let wasm_loader = html_utils.log_group_collapsed(download_info, async () => {
         let wasm_glue_js = await wasm_glue_fetch.text()
         let wasm_glue = Function('let exports = {};' + wasm_glue_js + '; return exports')()
-        let imports = wasm_glue.wasm_imports()
+        let imports = await wasm_glue.wasm_imports()
```
Comment (wdanilo) [inline app/ide-desktop/lib/content/src/index.ts:102]: ❤️

**Review by wdanilo — APPROVED** [review]

File: app/gui/analytics/Cargo.toml [hunk 4]
```diff
@@ -6,4 +6,4 @@ edition = "2021"
 
 [dependencies]
 js-sys = { version = "0.3.28" }
-wasm-bindgen = { version = "=0.2.58", features = ["nightly", "serde-serialize"] }
+wasm-bindgen = { version = "0.2.58", features = ["nightly", "serde-serialize"] }
```
Comment (farmaazon) [inline app/gui/analytics/Cargo.toml:9]: I prefer to keep 0.2.78 everywhere.

**Review by farmaazon — APPROVED** [review]

**Review by 4e6 — APPROVED** [review]

**Pull request merged**